File: src/main/scala/poc/ReadGeotrellis.scala
package poc

import java.net.URI
import astraea.spark.rasterframes._
import org.apache.spark.sql._

object ReadGeotrellis {
  val catalogUri = new URI("file:/data/workspace/geotrellis-landsat-tutorial/data/1k_tiles")
  
  def main(args: Array[String]): Unit = {
  
  implicit val spark = SparkSession.builder().
  master("local").appName("RasterFrames").
  config("spark.ui.enabled", "false").
  getOrCreate().
  withRasterFrames
  
  import astraea.spark.rasterframes.datasource.geotrellis._
  import java.io.File
  
  val catalog = spark.read.geotrellisCatalog(catalogUri)
  catalog.printSchema


  }
}

File: python/read-geotrellis.py
import geopyspark
import numpy

from pyspark.sql import SparkSession
from pyrasterframes import *

conf = geopyspark.geopyspark_conf(appName="POC")
session = SparkSession.builder.config(conf=conf).getOrCreate().withRasterFrames()
 

uri = "file:/data/workspace/geotrellis-landsat-tutorial/data/1k_tiles/"
layer_name = "landsat1K"

layer = geopyspark.query(uri=uri, layer_name=layer_name, layer_zoom=0)

rf = layer.to_rasterframe(3)
rf.show()
